base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -3995,3 +4115,76 @@ fn get_correct_accumulated_difficulty() -> Vec<(u64, U512)> {
     #[cfg(not(any(tari_target_network_mainnet, tari_target_network_nextnet)))]
     vec![]
 }
+
+fn get_correct_accumulated_difficulty_v5() -> Vec<(u64, U512)> {
+    #[cfg(tari_target_network_mainnet)]
+    {
+        vec![
+            // We have known bad values at the next height, so this one can be a good reference point
+            (
+                15224,
```

Comment:
we dont need this fine grained loop, rough estimates will be fine as we will fix these in the background. And it should be quick.